### PR TITLE
SQL: Adapt the limitations page to the new "fields" API usage

### PR DESCRIPTION
**Pull request by astefan**
After https://github.com/elastic/elasticsearch/pull/68802 was merged, this section of the SQL documentation needed an update.

### CODE REVIEW

File: docs/reference/sql/limitations.asciidoc [hunk 1]
```diff
@@ -196,22 +196,12 @@ indexed with some loss of precision from the original values (4.190951585769653E
...
-
-[discrete]
-[[fields-from-docvalues]]
-=== Retrieving from `docvalue_fields`
-
-When the number of columns retrievable from `docvalue_fields` is greater than the configured <<dynamic-index-settings,`index.max_docvalue_fields_search` setting>>
-the query will fail with `IllegalArgumentException: Trying to retrieve too many docvalue_fields` error. Either the mentioned {es}
-setting needs to be adjusted or fewer columns retrievable from `docvalue_fields` need to be selected.
+{es-sql} retrieves columns' values using <<search-fields-param,`_search` "fields" parameter>> and, as such, any
+limitations this parameter has, will be visible in {es-sql} queries, as well. For example, if `_source` is disabled
```
Comment (jrodewig) [inline docs/reference/sql/limitations.asciidoc:203]:
```suggestion
{es-sql} retrieves column values using the <<search-fields-param,search API's
`fields` parameter>>. Any limitations on the `fields` parameter also apply to
{es-sql} queries. For example, if `_source` is disabled
```

File: docs/reference/sql/limitations.asciidoc [hunk 2]
```diff
@@ -196,22 +196,12 @@ indexed with some loss of precision from the original values (4.190951585769653E
 Therefore calling `ST_Z` function in the filtering, grouping or sorting will return `null`.
 
 [discrete]
-[[fields-from-source]]
-=== Retrieving from `_source`
+[[using-fields-api]]
+=== Retrieving using "fields" search parameter
```
Comment (jrodewig) [inline docs/reference/sql/limitations.asciidoc:200]:
```suggestion
=== Retrieving using the `fields` search parameter
```

**Review by jrodewig — APPROVED** [review]
LGTM. I left two minor nits that are non-blocking. Thanks @astefan!

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**